### PR TITLE
Swift 3 support

### PR DESCRIPTION
**Pull request by mikeger**
- Keychain is broken on Simulator
- Having full module optimization revelaed that we used same-named enum in two classes (`Trace`)

### CODE REVIEW

File: Source/Background/BackgroundActivityFactory.swift [hunk 1]
```diff
@@ -19,33 +19,33 @@
 
 private var _instance : BackgroundActivityFactory? = BackgroundActivityFactory() // swift automatically dispatch_once make this thread safe
 
-@objc public class BackgroundActivityFactory: NSObject {
+@objc open class BackgroundActivityFactory: NSObject {
```
Comment (daehn) [inline Source/Background/BackgroundActivityFactory.swift]: Could be `public`

Comment (mikeger) [inline Source/Background/BackgroundActivityFactory.swift]: No, it is subclassed in SE: https://github.com/search?q=org%3Awireapp+BackgroundActivityFactory&type=Code

Comment (daehn) [inline Source/Background/BackgroundActivityFactory.swift]: Ahh to bad

**Comment by daehn** [comment]
While we are making breaking changes it would be nice to change the case of these methods containing `HTTPstatus` to `HTTPStatus`: https://github.com/wireapp/wire-ios-transport/blob/develop/Source/Public/ZMTransportResponse.h#L49 as they are currently auto imported to Swift as `httPsttatus` which looks very odd.

**Comment by mikeger** [comment]
@daehn updated + added nullability declarations

**Comment by daehn** [comment]
Please merge when ✅

**Comment by mikeger** [comment]
✅

File: Tests/Source/Authentication/ZMAccessTokenHandlerTests.m [hunk 2]
```diff
@@ -56,9 +56,9 @@ @implementation ZMAccessTokenHandlerTests
 
 - (void)setUp {
     [super setUp];
-    
+#if !TARGET_OS_IPHONE && TARGET_IPHONE_SIMULATOR
```
Comment (billypchan) [inline Tests/Source/Authentication/ZMAccessTokenHandlerTests.m]: Is there any reason for not writing to keychain on Simulator?

**Pull request merged**